build-gluster-org/scripts/generic_package.sh

```diff
@@ -0,0 +1,200 @@
+        mirror="http://ftp.us.debian.org/debian/"
+	case ${series} in
+	  "3.12")
+    	    debuild_key=${debuild_keys[0]}
+    	    pbuild_key=${pbuild_keys[0]}
+    	    ;;
+  	  "4.0")
+            debuild_key=${debuild_keys[1]}
+            pbuild_key=${pbuild_keys[1]}
+            ;;
+          "4.1")
+            if [ "$flavor" == "stretch" ]; then
+                 debuild_key=${pbuild_keys[2]}
+            else
+                 debuild_key=${debuild_keys[2]}
+            fi
+            pbuild_key=${pbuild_keys[2]}
+            ;;
+          "5" | "6" | "7")
+            if [ "$flavor" == "stretch" ]; then
+                 debuild_key=${pbuild_keys[3]}
+            else
+                 debuild_key=${debuild_keys[3]}
+            fi
+            pbuild_key=${pbuild_keys[3]}
+        esac
+else
+	echo "Exiting: OS should be debian or ubuntu. Please provide the right one"
+	exit
+fi
+
+mkdir ${os}-${flavor}-Glusterfs-${version}
+
+cd ${os}-${flavor}-Glusterfs-${version}
+
+mkdir build packages
+
+echo "Building glusterfs-${version}-${release} for ${flavor}"
+
+cd build
+
+TGZS=(`ls ~/glusterfs-${version}-?-*/build/glusterfs-${version}.tar.gz`)
+echo ${TGZS[0]}
+
+if [ -z ${TGZS[0]} ]; then
+        echo "wget https://download.gluster.org/pub/gluster/glusterfs/${series}/${version}/glusterfs-${version}.tar.gz"
+        wget https://download.gluster.org/pub/gluster/glusterfs/${series}/${version}/glusterfs-${version}.tar.gz
+else
+        echo "found ${TGZS[0]}, using it..."
+        cp ${TGZS[0]} .
+fi
+
+echo "Creating link file.."
+ln -s glusterfs-${version}.tar.gz glusterfs_${version}.orig.tar.gz
+
+echo "Untaring.."
+tar xpf glusterfs-${version}.tar.gz
+
+# Changelogs needed for building are maintained in a separate repo.
+# the repo has to be clone and updated properly so we can copy the changelogs so far.
+
+echo "Cloning the glusterfs-debian repo"
+git clone http://github.com/gluster/glusterfs-debian.git
+
+cd glusterfs-debian/
+
+git checkout -b ${flavor}-${series}-local origin/${flavor}-glusterfs-${series}
+
+if [ "$os" == "ubuntu" ]; then
+	sed -i "1i glusterfs (${version}-${os}1~${flavor}1) ${flavor}; urgency=medium\n\n * GlusterFS ${version} GA\n\n -- GlusterFS GlusterFS deb packages <deb.packages@gluster.org>  `date +"%a, %d %b %Y %T %z"`\n" debian/changelog
+elif [ "$os" == "debian" ]; then
+	sed -i "1i glusterfs (${version}-1) ${flavor}; urgency=low\n\n * GlusterFS ${version} GA\n\n -- Gluster Packager <glusterpackager@download.gluster.org>  `date +"%a, %d %b %Y %T %z"`\n" debian/changelog
+fi
+
+git commit -a -m "Glusterfs ${version} G.A (${flavor})"
+
+echo "Copying Changelog to source"
+cp -a debian ../glusterfs-${version}/
+
+echo "Building source package.."
+cd ../glusterfs-${version}
+debuild -S -sa -k${debuild_key}
+
+echo "Uploading the packages.."
+if [ "$os" == "ubuntu" ]; then
+        cd ..
+        dput ppa:gluster/glusterfs-${series} glusterfs_${version}-${os}1~${flavor}1_source.changes
```

Comment:
Does it requires some authentication ? I do not see how it is done.

Reply:
No, AIFIK. @kalebskeithley any more insights?

Reply:
Well, if could be without authentication, but then, we need some kind of signature, cause I do not think anyone can push debian package for us, no ?

Reply:
yes, we would need to follow https://help.launchpad.net/Packaging/PPA/Uploading to get required authentiacation

Reply:
The documentation do not tell much, but if that use ssh, then we need to have the key available to the builder, which is not declared in the job yaml file. There is example here: https://github.com/gluster/build-jobs/blob/master/build-gluster-org/jobs/centos7-regression.yml#L64 for adding a credential

Reply:
The  OpenPGP keys are used for signing as described [here](https://help.launchpad.net/YourAccount/ImportingYourPGPKey#:~:text=Publishing%20your%20key%20to%20a%20keyserver,-Step%201%3A%20Open&text=Choose%20the%20Sync%20button.,import%20key%20into%20into%20Launchpad.)
The above link also shows how a new key can be created and added to launcpad account.

$ gpg --list-keys on the machine rhs-vm-17.storage-dev.lab.eng.bos.redhat.com under the user glusterpackager, shows the already existing keys. (documented the same in the mojo doc draft: let me know if any more information is needed to be documented)

Reply:
yes, we can store the key as secret in the job. We can add it in jenkins machine, I can help in getting it from  rhs-vm-17.storage-dev.lab.eng.bos.redhat.com